server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -938,8 +939,7 @@ private <T> void termsAggregator(
 
                     // Testing actual results, by first building a map of expected key->count because key order can change
                     HashMap<Object, Long> expectedCounts = new HashMap<>();
-                    for (int i = 0; i < size; i++) {
-                        Map.Entry<T, Integer> expected = expectedBuckets.get(i);
+                    for (Map.Entry<T, Integer> expected : expectedBuckets) {
```

Comment:
Ah, because of possible order changes we need to use the entire expected set to compare against!